metals/src/main/scala/scala/meta/internal/metals/ScalaVersionSelector.scala

```diff
@@ -0,0 +1,27 @@
+package scala.meta.internal.metals
+
+class ScalaVersionSelector(
+    userConfig: () => UserConfiguration,
+    buildTargets: BuildTargets
+) {
+
+  def fallbackScalaVersion(allowScala3: Boolean): String = {
+    val selected = userConfig().fallbackScalaVersion match {
+      case Some(v) => v
+      case None =>
+        buildTargets.all.toList
+          .map(_.scalaInfo.getScalaVersion)
+          .sorted
+          .lastOption
+          .getOrElse(BuildInfo.scala212)
+    }
+
+    // ammonite doesn't support Scala3 yet
+    if (!allowScala3 && ScalaVersions.isScala3Version(selected))
+      BuildInfo.scala213
+    else if (ScalaVersions.isSupportedScalaVersion(selected))
+      selected
+    else
+      ScalaVersions.recommendedVersion(selected)
```

Comment:
We should show message in this case, that the selected version is not supported.

Reply:
Added warning about that on `workspace/didChangeConfiguration`